packages/block-library/src/columns/editor.scss

```diff
@@ -89,27 +89,43 @@
 
 			// Beyond mobile, allow 2 columns.
 			@include break-small() {
-				flex-basis: 50%;
+				// Size should be half the available width, minus the margins added below.
+				// In the editor, we also need to account for additional block padding.
+				flex-basis: calc(50% - #{$grid-size-large * 2} - #{$block-padding * 2});
```

Comment:
I took a stab at fixing the margin issue I noted. When I reverted this back to `flex-basis: 50%;` (editor only), I got things working:

<img width="813" alt="screenshot 2018-12-17 at 09 03 05" src="https://user-images.githubusercontent.com/1204802/50073958-ab4d9f00-01da-11e9-80f2-81b2d07f3b40.png">

Keep in mind that the editor uses a lot of negative margins left and right so as to make available a selectable/hoverable space left and right of blocks. This might be why the additional grid stuff isn't necessary here. Am I missing something? 

This is with `flex-basis: 50%;`:

<img width="813" alt="screenshot 2018-12-17 at 09 03 05" src="https://user-images.githubusercontent.com/1204802/50074026-dc2dd400-01da-11e9-8acb-341f642c53fd.png">

<img width="918" alt="screenshot 2018-12-17 at 09 04 47" src="https://user-images.githubusercontent.com/1204802/50074030-df28c480-01da-11e9-89d2-3ea6c23a985d.png">

Again, I have no doubt that this is necessary on the front-end, but I think it's what's causing issues in the editor.

Reply:
Hm. When I change this back to `flex-basis: 50%;`, this is what I get at the Small (600px) breakpoint.

Did you have to make any other changes?

![image](https://user-images.githubusercontent.com/1050936/50093634-e057f280-01d6-11e9-83aa-c6bcb2d05a1a.png)

Reply:
Ah good call. But if you change "break-medium" back to "break-small" here: https://github.com/WordPress/gutenberg/pull/12816/files#diff-49ac41bc72e24e343a451fe254801c0fR38 — it works again:

![screenshot 2018-12-17 at 16 51 43](https://user-images.githubusercontent.com/1204802/50098373-0e135a80-021c-11e9-9798-12518e686aad.png)

Reply:
At the small (2-column) breakpoint… I think as long as there's a margin b/w the columns, then `flex-basis` needs to account for that doesn't it?  50% plus any margin will always wrap after the first column, right?

Reply:
Possibly. But keep in mind that there is negative margins applied to columns in the _editor only_, to ensure that this looks right. Keep in mind the markup in the editor is different from the markup on the frontend — there are a lot more divs in the editor to enable the various tools for each block. So until we have access to something like `display: contents;` there's likely always going to be a discrepancy between the frontend and backend. 

Did you try flex-basis: 50%? For me it seemed to fix the issue of the extra right margin, but there could be situations I haven't accounted for there.

Reply:
> For me flex-basis:50%; fixes the extra margin at the medium breakpoint, but only at the expense of breaking 2-column layout at the small breakpoint.

This only works if you _also_ set the wrap property to nowrap as noted in https://github.com/WordPress/gutenberg/pull/12816#discussion_r242202364.

But I'm going to take a look at your latest push, see if that fixes it!